src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -46,10 +43,17 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 {
                     INamedTypeSymbol? baseType = method.ContainingType.BaseType;
                     // If the base type does not have default constructor.
-                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.Where(c => c.Parameters.IsEmpty).Count() == 0)
+                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.All(c => !c.Parameters.IsEmpty))
                     {
-                        ConstructorDeclarationSyntax declaration = (ConstructorDeclarationSyntax)syntaxGenerator.Declaration(method);
-                        return declaration.WithInitializer(GenerateBaseConstructorInitializer(baseType.Constructors));
+                        IOrderedEnumerable<IMethodSymbol> baseTypeConstructors = baseType.Constructors
+                            .Where(c => !c.GetAttributes().Any(a => a.IsObsoleteWithUsageTreatedAsCompilationError()))
+                            .OrderBy(c => c.Parameters.Count());
```

Comment:
An analyzers inside VS suggests to use Length instead of Count():

```suggestion
                            .OrderBy(c => c.Parameters.Length);
```

Reply:
updated